File: vol3/vol3-java-examples/src/main/java/com/heatonresearch/aifh/examples/classic/logic/Connection.java
package com.heatonresearch.aifh.examples.classic.logic;

/**
 * Connection between two neurons.
 */
public class Connection {
    /**
     * The weight of the connection.
     */
    private double weight;

    /**
     * The parent/source.
     */
    private Neuron parent;

    /**
     * Create a connection.
     * @param weight The weight.
     * @param parent The parent/source neuron.
     */
    public Connection(double weight, Neuron parent) {
        this.weight = weight;
        this.parent = parent;
    }

    /**
     * @return The weight.
     */
    public double getWeight() {
        return weight;
    }

    /**
     * Set the weight.
     * @param weight The weight.
     */
    public void setWeight(double weight) {
        this.weight = weight;
    }

    /**
     * @return The parent neuron.
     */
    public Neuron getParent() {
        return parent;
    }

    /**
     * Set the parent neuron.
     * @param parent The parent neuron.
     */
    public void setParent(Neuron parent) {
        this.parent = parent;
    }
}

File: vol3/vol3-csharp-examples/AIFH-Vol3/Examples/Classic/Logic/Connection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AIFH_Vol3.Examples.Classic.Logic
{
    /// <summary>
    /// Connection between two neurons.
    /// </summary>
    public class Connection
    {
        /// <summary>
        /// The weight of the connection.
        /// </summary>
        public double Weight { get; set; }

        /// <summary>
        /// The parent/source.
        /// </summary>
        public INeuron Parent { get; set; }
        
        /// <summary>
        /// Create a connection. 
        /// </summary>
        /// <param name="weight">The weight.</param>
        /// <param name="parent">The parent/source neuron.</param>
        public Connection(double weight, INeuron parent)
        {
            Weight = weight;
            Parent = parent;
        }
        
    }
}
